src/Symfony/Component/DependencyInjection/Attribute/AutowireInline.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+
+/**
+ * Allows inline service definition at the class level.
+ *
+ * @author Ismail Özgün Turan <oezguen.turan@dadadev.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireInline extends Autowire
```

Comment:
`AutowireInline` looks pretty cryptic, any newcomer wouldn't be able to get what the attribute by just looking at its name nor its description as it currently is. I think we either need find a super self-explanatory name (I've no clue) or extend the description so that it tells what's the purpose of the attribute and when it should be used.

Reply:
Naming things... :)
"inline" refers to "inline_service()" in the PHP-DSL

AutowireInlineService? but verbose
AutowireNew? AutowireObject? AutowireInstance? or keep AutowireInline?

of course, a top notch description is also desirable

Reply:
I added a better description [here](https://github.com/symfony/symfony/compare/dd16390a5290a9a08b2c997fa9d959296bb6f0e0..c94322321acafdbaf32044e58b213223645497ce) feel free to request further suggestments to it :+1:.